avalon/io.py

```diff
@@ -220,6 +220,24 @@ def decorated(*args, **kwargs):
     return decorated
 
 
+def auto_reconnect(f):
+    """Handling auto reconnect in 3 retry times"""
+    @functools.wraps(f)
+    def decorated(*args, **kwargs):
+        error = None
+        for retry in range(3):
+            try:
+                return f(*args, **kwargs)
+            except pymongo.errors.AutoReconnect as e:
+                error = e
```

Comment:
Do we need this `error = e`, couldn't you just ignore storing the error as variable and just in in the else statement `raise` without a variable to raise the last error that happened? Or am I misinterpreting what this is trying to do?

Reply:
Oh ! yes, you are right. It's redundant !